src/main/java/org/spongepowered/api/item/recipe/smelting/SmeltingRecipe.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.recipe.smelting;
+
+import com.google.common.collect.ImmutableCollection;
+import com.google.common.collect.ImmutableList;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.item.ItemType;
+import org.spongepowered.api.item.inventory.ItemStack;
+import org.spongepowered.api.item.inventory.ItemStackSnapshot;
+import org.spongepowered.api.item.recipe.Recipe;
+import org.spongepowered.api.util.ResettableBuilder;
+
+/**
+ * This represents a {@link Recipe} to be used in a Furnace.
+ * @see Recipe
+ */
+public interface SmeltingRecipe extends Recipe {
+
+    /**
+     * Creates a new {@link Builder} to build a {@link SmeltingRecipe}.
+     *
+     * @return The new builder
+     */
+    static Builder builder() {
+        return Sponge.getRegistry().createBuilder(Builder.class);
+    }
+
+    /**
+     * Gets the experience given when this recipe is made.
+     *
+     * @return the experience given when this recipe is made
+     */
+    double getExperience();
+
+    /**
+     * Gets the item needed to make this recipe.
+     * (Minecraft codes it so that it only takes one item)
+     *
+     * @return item needed to make this recipe
+     */
+    ItemStackSnapshot getIngredient();
+
+    /**
+     * Gets the result of this {@link Recipe}.
+     *
+     * @return the result of this {@link Recipe}
+     */
+    ItemStackSnapshot getResult();
+
+    default ImmutableCollection<ItemStackSnapshot> getResults() {
```

Comment:
@Maxqia Yes, `@Override` works here. You should always annotate the methods appropriately so it's plain and clear that the method doesn't originate from this interface/class and is an overridden method. Everything in Sponge follows this rule.